tools/tsp-client/src/typespec.ts

```diff
@@ -53,3 +54,34 @@ export async function compileTsp({
     Logger.success("generation complete");
   }
 }
+
+// TODO add emitter options
+export async function runTspCompile({
+  tempDir,
+  mainFilePath,
+  emitter,
+  emitterOptions,
+}: {
+  tempDir: string;
+  mainFilePath: string;
+  emitter: string;
+  emitterOptions: string;
+}): Promise<void> {
+  return new Promise((resolve, reject) => {
+    const git = spawn("tsp", ["compile", mainFilePath, `--emit=${emitter}`, emitterOptions], {
```

Comment:
why are we shelling out to tsp instead of invoking the compiler directly?

another shell injection attack here as well